pkg/typesystem/typesystem_create_test.go

```diff
@@ -0,0 +1,426 @@
+package typesystem
+
+import (
+	"testing"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/stretchr/testify/require"
+
+	"github.com/openfga/openfga/internal/condition"
+	"github.com/openfga/openfga/pkg/testutils"
+)
+
+// usersetsEquals tests that the two *openfgav1.Usersets
+// values provided are equal. This is a deep equality check.
+func usersetsEquals(t *testing.T, a, b *openfgav1.Usersets) {
+	if a == nil || b == nil {
+		require.Equal(t, a, b)
+		return
+	}
```

Comment:
**(suggestion):** May I suggest something of sorts instead:

```go
if a == nil && b == nil {
  return
}

require.NotNil(t, a)
require.NotNil(t, b)
```

🤔 I personally find this easier to understand and more readable. I feel like this expresses the intent a bit better, even if with more lines of code.

---

**(suggestion):** Given we're adding a lot of helper assert funcs, I'd suggest we mark these as helper funcs through the usage of `t.Helper()`, so that if an error or failure occurs inside this helper function, the line of code reported in the test output will be the line in the calling function (the test function), not the line inside the helper function itself. This makes the test output more useful because it points you directly to the line in the test that caused the failure.

---

⚠️ The suggestions above apply to multiple funcs in this PR.

Reply:
1. I'm fine with making the suggested change, as the result is the same. The benefit is subjective, but I tend to lean on making code easier for others to read.
2. Using `t.Helper()` is a good idea here, thank you.